test/puppetlabs/ring_middleware/core_test.clj

```diff
@@ -278,7 +279,7 @@
          :endpoint      "/hello-proxy"
          :target-endpoint "/hello"}
         (let [response (http-get "http://localhost:10000/hello-proxy/wrong-host")]
-          (is (= (:status response 502))))))
+          (is (= (:status response) 500)))))
```

Comment:
Thanks for fixing this test! I think this functionality might actually be broken (my fault, I wrote this test poorly).  I did a little bit of testing, and from what I can tell, this test is passing because there's nothing running on `localhost:4`, rather than because `localhost:4` is the wrong host. 

I definitely don't think this needs to be fixed in this PR, I'll open a ticket for it.